rpc-client.py

```diff
@@ -1,118 +1,106 @@
-
-
-def rpc_call(method_name, **kwargs):
-    payload = {
-        "method": method_name,
-        "params": kwargs,
-        "jsonrpc": "2.0",
-        'id': 0,
-    }
-
-    res = requests.post(url, data=json.dumps(payload), headers=headers)
-    response = res.json()
-
-    return response
-
-
-def get_all():
-    payload = {
-        "method": "App.get_all",
-        "params": None,
-        "jsonrpc": "2.0",
-        'id': 0,
-    }
-
-    response = requests.post(
-        url, data=json.dumps(payload), headers=headers).json()
-
-    return response
-    # assert response["result"] == "hello world!"
-    # assert response["jsonrpc"]
-    # assert response["id"] == 0
-
-
-def get(name='test-item #1'):
-    payload = {
-        "method": "App.get",
-        "params": {'name': name},
-        "jsonrpc": "2.0",
-        'id': 0,
-    }
-
-    response = requests.post(
-        url, data=json.dumps(payload), headers=headers).json()
 
-    return response
-
-
-def open_vault(username, password, vault_name):
-    payload = {
-        "method": "App.open_vault",
-        "params": {
-            'username': username,
-            'password': password,
-            'vault_name': vault_name
-        },
-        "jsonrpc": "2.0",
-        'id': 0,
-    }
-
-    response = requests.post(
-        url, data=json.dumps(payload), headers=headers).json()
-
-    return response
-
-
-def fake_open_vault(username='User 1', password='Some secret',
-                    vault_name='my-vault'):
-    payload = {
-        "method": "App.open_vault",
-        "params": {
-            'username': username,
-            'password': password,
-            'vault_name': vault_name
-        },
-        "jsonrpc": "2.0",
-        'id': 0,
-    }
-
-    fake_headers = {
-        'content-type': 'application/json',
-        'token': 'WRONG TOKEN'
-    }
-
-    response = requests.post(
-        url, data=json.dumps(payload), headers=fake_headers).json()
-
-    return response
-
-
-def getTokenAndPort():
-    p = Popen(['python', 'rpc-server.py'], stdout=PIPE)
-    token = p.stdout.readline().decode('ascii').strip()
-    port = p.stdout.readline().decode('ascii').strip()
-    return token, port
+class RPCClient:
+    def __init__(self):
+        self._headers = {'content-type': 'application/json'}
+        self._server_started = False
+
+    def start_server(self):
+        """Start server and use it for API calls."""
+        self._server = Popen(['python', 'rpc-server.py'], stdout=PIPE)
+        out = self._server.stdout
+        token = out.readline().decode('ascii').strip()
+        port = out.readline().decode('ascii').strip()
+
+        self._url = "http://localhost:{port}/api/v1".format(port=port)
+        self._headers['token'] = token
+
+        # Wait until the server starts
+        sleep(2)
+
+        print('Server started')
+        print("Token: {token}".format(token=token))
+        print("Port: {port}".format(port=port))
+        self._server_started = True
```

Comment:
@ivanalejandro0, we should start the server only if the flag `_server_started` is `False`. Thus, it avoids the  user to start multiple RPC servers. It's not a show stopper at this stage, you can opt to not add it right now.